File: Library.ViewModels/Book/PostBookViewModel.cs
using Library.ViewModels.Author;
using Library.ViewModels.Publication;
using Library.ViewModels.Publisher;
using System;
using System.Collections.Generic;

namespace Library.ViewModels.Book
{
    public class PostBookViewModel : PublicationViewModel
    {
        public List<AuthorViewModel> Authors { get; set; }

        public DateTime DateOfPublishing { get; set; }

        public List<PublisherViewModel> Publishers { get; set; }

        public PostBookViewModel()
        {
            Authors = new List<AuthorViewModel>();
            Publishers = new List<PublisherViewModel>();
        }
    }
}

File: Library.WEB/src/app/viewModels/book/postBookViewModel.ts
import { GetPublisherViewItem } from '../publisher/getPublisherViewItem';
import { GetAuthorViewItem } from '../author/getAuthorViewItem';

export class PostBookViewModel {
    id: number
    name: string
    authors: Array<GetAuthorViewItem>
    dateOfPublishing: Date
    type: string
    publishers: Array<GetPublisherViewItem>
}
